codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractDiskHttpData.java

```diff
@@ -300,15 +300,19 @@ public ByteBuf getChunk(int length) throws IOException {
         }
         int read = 0;
         ByteBuffer byteBuffer = ByteBuffer.allocate(length);
-        while (read < length) {
-            int readnow = fileChannel.read(byteBuffer);
-            if (readnow == -1) {
-                fileChannel.close();
-                fileChannel = null;
-                break;
-            } else {
-                read += readnow;
+        try {
+            while (read < length) {
+                int readnow = fileChannel.read(byteBuffer);
+                if (readnow == -1) {
+                    fileChannel.close();
+                    fileChannel = null;
+                    break;
+                } else {
+                    read += readnow;
+                }
             }
+        } finally {
+            fileChannel.close();
```

Comment:
this will throw a NPE if `readnow == -1` is true. you will need to add a null check here